src/transports/sentry_transport_winhttp.c

```diff
@@ -110,6 +111,21 @@ static int
 sentry__winhttp_transport_shutdown(uint64_t timeout, void *transport_state)
 {
     sentry_bgworker_t *bgworker = (sentry_bgworker_t *)transport_state;
+    winhttp_bgworker_state_t *state = sentry__bgworker_get_state(bgworker);
+
+    if (state->connect) {
+        WinHttpCloseHandle(state->connect);
+        state->connect = NULL;
+    }
+    if (state->session) {
+        WinHttpCloseHandle(state->session);
+        state->session = NULL;
+    }
+    if (state->request) {
+        WinHttpCloseHandle(state->request);
+        state->request = NULL;
+    }
+
     return sentry__bgworker_shutdown(bgworker, timeout);
```

Comment:
Thanks for the explanation why the ordering is important. It might be a good idea to put that as a comment in the code.

I also saw that all the tests are failing, I think in that case also the ordering is important.
The main thread will ask the background thread to shut down here according to the timeout. I think you have to move all your close calls *after* this call here, because you *do* want to wait for the given timeout, instead of just throwing away all the requests that are currently in flight.